python-api/app/integrations/openrouter/clients.py

```
@@ -0,0 +1,209 @@
+                    try:
+                        error_data = json.loads(response_text)
+                        raise OpenRouterAPIError(
+                            message=error_data.get("error", {}).get("message", "Unknown error"),
+                            status_code=response.status,
+                            error_data=error_data
+                        )
+                    except json.JSONDecodeError:
+                        raise OpenRouterAPIError(
+                            message=response_text,
+                            status_code=response.status
+                        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)